src/components/FeatureViews/Scheduler/Scheduler.js

```diff
@@ -47,39 +52,80 @@ class Scheduler extends Component {
             "viewValue": "timeGridDay",
             "filterValue": "class",
             "resourceFilterValue": "R",
-            "calendarFilterValue": "C",
+            "calendarFilterValue": "T",
             "calendarIcon": true,
             "resourceIcon": false,
             "filterTypeCalendar": "",
+            "timeShift": 0,
+        };
+        this.calendarViewToFilterVal = {
+            "timeGridDay": "day",
+            "timeGridWeek": "week",
+            "dayGridMonth": "month",
+        };
+        this.viewOptions = {
+            "T": "tutoring",
+            "C": "class",
         };
-
     }
 
     calendarComponentRef = React.createRef();
 
-    componentWillMount() {
-        this.setState({
-            "calendarEvents": this.getEvents(),
-        });
 
-        this.getFirstFilteredList();
-    }
 
     componentDidMount() {
+        this.props.calendarActions.fetchSessions({
+            config: {
+                params: {
+                    time_frame: "day",
+                    view_option: "tutoring",
+                    timeShift: this.state.timeShift,
+                }
+            }
+        });
+        this.props.courseActions.fetchCourses();
+        this.props.userActions.fetchInstructors();
         this.setState({
             "currentDate": this.currentDate(),
         });
+        console.log(this.calendarComponentRef)
     }
 
-    // helper function filtering by Class and setting the calendarEvents state, Which is read by Full calendar
-    getFirstFilteredList = () => {
-        const items = this.getEvents();
-        const newEvents = items.filter((item) => item.type === "C");
-        this.setState((prevState) => (
-            {
-                "calendarEvents": prevState.calendarEvents = newEvents,
-            }));
-        sessionStorage.setItem("calendarEvent", JSON.stringify(newEvents));
+    componentDidUpdate(prevProps, prevState, snapshot) {
+        if (this.props !== prevProps && this.props.sessions !== "" &&
+            !(Object.entries(this.props.courses).length === 0 &&
+                this.props.courses.constructor === Object) &&
```

Comment:
What exactly are you checking for here? You may not need to compare the constructor.